spec/checker_spec.lua

```diff
@@ -55,20 +55,21 @@ describe("Scope analysis: ", function()
             "variable 'x' is not declared")
     end)
 
-    it("forbids multiple toplevel declarations with the same name", function()
+    it("forbids multiple toplevel declarations with the same name for exported functions", function()
         assert_error([[
-            local function f() end
-            local function f() end
+            export function f() end
+            export function f() end
         ]],
-            "duplicate toplevel declaration for 'f'")
+            "duplicate export 'f', previous one at line 1")
     end)
 
-    it("forbids multiple toplevel variable declarations with the same name", function()
-        assert_error([[
-            local a, a = 1, 2
-        ]],
-            "duplicate toplevel declaration for 'a'")
-    end)
+    -- Shouldn't this be removed?
+    -- it("forbids multiple toplevel variable declarations with the same name", function()
+    --     assert_error([[
+    --         local a, a = 1, 2
+    --     ]],
+    --         "duplicate toplevel declaration for 'a'")
+    -- end)
```

Comment:
Instead of removing it, I would move this test to coder_spec, the "Corner cases of scoping" section.

Please also add a test there to check what happens if you have two local functions with the same name.